include/RAJA/policy/cuda/forall.hpp

```diff
@@ -47,76 +47,133 @@
 
 namespace RAJA
 {
-
 namespace policy
 {
-
 namespace cuda
 {
 
 namespace impl
 {
 
-/*!
- ******************************************************************************
- *
- * \brief calculate gridDim from length of iteration and blockDim
- *
- ******************************************************************************
- */
-RAJA_INLINE
-cuda_dim_t getGridDim(cuda_dim_member_t len, cuda_dim_t blockDim)
+template<typename IterationMapping, typename IterationGetter, typename UniqueMarker>
+struct ForallDimensionCalculator;
+
+template<named_dim dim, int BLOCK_SIZE, int GRID_SIZE, typename UniqueMarker>
+struct ForallDimensionCalculator<::RAJA::iteration_mapping::Direct,
+                                 ::RAJA::cuda::IndexGlobal<dim, BLOCK_SIZE, GRID_SIZE>,
+                                 UniqueMarker>
 {
-  cuda_dim_member_t block_size = blockDim.x * blockDim.y * blockDim.z;
+  static_assert(BLOCK_SIZE >= 0, "block size may not be ignored with forall");
```

Comment:
Does this work with either equal to zero?

Reply:
I believe 0 means unspecified so it can be determined at runtime. But that is not clear in this check.

Reply:
Maybe make the assert comment more explicit for cases < 0, = 0, and > 0?

Reply:
Made these tests for explicit.